cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -223,27 +249,37 @@ func (m *McmManager) GetMachineDeploymentForMachine(machine *Ref) (*MachineDeplo
 	}, nil
 }
 
-//Cleanup does nothing at the moment.
-//TODO: Enable cleanup method for graceful shutdown.
+// Refresh refreshes the lister caches
+func (m *McmManager) Refresh() error {
+	var stopCh <-chan struct{}
+	if !cache.WaitForCacheSync(stopCh, m.machineDeploymentSynced, m.machineSetSynced, m.machineSynced, m.machineClassSynced, m.azureMachineClassSynced, m.awsMachineClassSynced) {
```

Comment:
Ideally, `WaitForCacheSync` should be called only once during startup and not during revery reconcile/main loop.
Maybe it is enough to do this [here](https://github.com/gardener/autoscaler/pull/73/files#diff-fc87dde41e1a7abff0b0bc24dbabc258ac3d594a346b8f01e020b9e7f9775440R172) once and then do nothing in this `Refresh` func because once cache is synced it will be kept up-to-date by the watch connections of the informers.

Reply:
Now checking only on [intialization here](https://github.com/gardener/autoscaler/pull/73/commits/a7d47db2a052a8b144e361afe5dae97e9aeea612)